auslib/db.py

```diff
@@ -935,6 +935,11 @@ def __init__(self, db, dialect, metadata, baseTable, conditions=("time", "uptake
             if col.primary_key or newcol.name == "base_data_version":
                 newcol.primary_key = False
                 newcol.nullable = True
+            # elif newcol.name in ('base_update_type', 'base_product', 'base_data', 'data_version'):
+                # newcol.nullable = True
+
+            # if newcol.name in ('base_name', 'base_permission', 'base_username'):
+                # newcol.nullable = False
```

Comment:
You shouldn't need to be hardcoding column names from specific baseTables here. My previous comment has some pointers on how to find these programatically instead.